java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java

```diff
@@ -354,36 +359,57 @@ private static void searchInFolder(List<TestItem> resultList, SearchTestItemPara
 
     private static void searchInPackage(List<TestItem> resultList, SearchTestItemParams params)
             throws JavaModelException {
-        final IPackageFragment packageFragment = resolvePackage(params.getUri(), params.getFullName());
-        if (packageFragment == null) {
+        final IJavaElement packageFragment = resolvePackage(params.getUri(), params.getFullName());
+        if (packageFragment == null || !(packageFragment instanceof IPackageFragment)) {
             return;
         }
 
-        for (final ICompilationUnit unit : packageFragment.getCompilationUnits()) {
+        for (final ICompilationUnit unit : ((IPackageFragment) packageFragment).getCompilationUnits()) {
             for (final IType type : unit.getTypes()) {
                 resultList.add(TestItemUtils.constructTestItem(type, TestLevel.CLASS));
             }
         }
     }
 
-    private static IPackageFragment resolvePackage(String uriString, String fullName) throws JavaModelException {
-        if (TestItemUtils.DEFAULT_PACKAGE_NAME.equals(fullName)) {
-            final IFolder resource = (IFolder) JDTUtils.findResource(JDTUtils.toURI(uriString),
+    private static IJavaElement resolvePackage(String uriString, String fullName) throws JavaModelException {
+        final IFolder resource = (IFolder) JDTUtils.findResource(JDTUtils.toURI(uriString),
                     ResourcesPlugin.getWorkspace().getRoot()::findContainersForLocationURI);
-            final IJavaElement element = JavaCore.create(resource);
-            if (element instanceof IPackageFragmentRoot) {
+        final IJavaElement element = JavaCore.create(resource);
+        if (element == null) {
+            return null;
+        }
+        if (element instanceof IPackageFragmentRoot) {
+            if (TestItemUtils.DEFAULT_PACKAGE_NAME.equals(fullName)) {
                 final IPackageFragmentRoot packageRoot = (IPackageFragmentRoot) element;
                 for (final IJavaElement child : packageRoot.getChildren()) {
                     if (child instanceof IPackageFragment && ((IPackageFragment) child).isDefaultPackage()) {
                         return (IPackageFragment) child;
                     }
                 }
             }
-        } else {
-            return JDTUtils.resolvePackage(uriString);
         }
 
-        return null;
+        return element;
+    }
+
+    /**
+     * Get all sub-packages whose name starts with the given package.
+     */
+    public static IPackageFragment[] getAllSubPackages(IJavaElement packageFragment) throws CoreException {
+        final List<IPackageFragment> result = new ArrayList<>();
+        if (packageFragment != null && packageFragment instanceof IPackageFragment) {
+            final IJavaElement[] packages = ((IPackageFragmentRoot) packageFragment.getParent()).getChildren();
+            for (final IJavaElement p : packages) {
+                if (!(p instanceof IPackageFragment)) {
```

Comment:
looks better in style?
```
if (p instanceof IPackageFragment) 
   if (p.getElementName().startsWith(packageFragment.getElementName()))
            result.add((IPackageFragment) p);
```

Reply:
Got the point, or maybe merge the if blocks into one?